### PR TITLE
[BEAM-204] Protect against empty stack traces

### PR DESCRIPTION
**Pull request by mshields822**
R: @lukecwik

### CODE REVIEW

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/UserCodeException.java [hunk 1]
```diff
@@ -62,13 +62,17 @@ private UserCodeException(Throwable t) {
...
    */
   private void truncateStackTrace(Throwable t) {
-
     StackTraceElement[] currentStack = Thread.currentThread().getStackTrace();
     StackTraceElement[] throwableStack = t.getStackTrace();
 
     int currentStackSize = currentStack.length;
     int throwableStackSize = throwableStack.length;
 
+    if (throwableStack.length == 0) {
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/UserCodeException.java]: throwableStackSize?

File: sdks/java/core/src/test/java/org/apache/beam/sdk/util/UserCodeExceptionTest.java [hunk 2]
```diff
@@ -96,6 +96,13 @@ public void testWrapIfReturnsSourceRuntimeExceptionWhenFalse() {
     assertEquals(runtimeException, wrapped);
   }
 
+  @Test
+  public void robustAgainstEmptyStackTrace() {
+    RuntimeException runtimeException = new RuntimeException("empty stack");
+    runtimeException.setStackTrace(new StackTraceElement[0]);
+    RuntimeException wrapped = UserCodeException.wrapIf(true, runtimeException);
```
Comment (tgroh) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/util/UserCodeExceptionTest.java]: This can just be UserCodeException.wrap

**Comment by tgroh** [comment]
Drive-by LGTM

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/UserCodeException.java (hunk 1 above)
Comment (mshields822) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/UserCodeException.java]: Done

File: sdks/java/core/src/test/java/org/apache/beam/sdk/util/UserCodeExceptionTest.java (hunk 2 above)
Comment (mshields822) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/util/UserCodeExceptionTest.java]: Done

**Comment by lukecwik** [comment]
LGTM

**Pull request closed without merging**